charts/kubernetes-agent/migrations.md

```diff
@@ -0,0 +1,38 @@
+# Migrations
+## Version 1 --> 2
+### Overview
+Version 2 of the Kubernetes-agent Helm chart was created to allow the agent to be used as a scalable worker
+by an OctopusDeploy instance.
+
+To support this change, deploymentTarget specific Values were moved under a deploymentTarget specific parent node in `values.yml`.
+
+This change in structure, demands a manual process to move your existing values to the new shape when upgrading from
+v1 to v2.
+
+Specifically, the following data items have been moved:
+* agent.targetName => agent.name
+* agent.targetEnvironments => agent.deploymentTarget.initial.environments
+* agent.defaultNamespace => agent.deploymentTarget.initial.defaultNamespace
+* agent.targetRoles => agent.deploymentTarget.initial.tags
+* agent.targetTenantTags => agent.deploymentTarget.initial.tenantTag
+* agent.targetTenants => agent.deploymentTarget.initial.tenants
+* agent.targetTenantedDeploymentParticipation => agent.deploymentTarget.initial.targetTenantedDeploymentParticipation
+* agent.scriptPods.image => agent.scriptPods.deploymentTarget.image (child-fields are unchanged)
```

Comment:
Can we do this as a table... Something like `From | To | Comments`, it'll just make it easier to grok